sdk/servicebus/azure-servicebus/azure/servicebus/_common/_connection_string_parser.py

```diff
@@ -0,0 +1,67 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse  # type: ignore
+
+from azure.servicebus.management._models import DictMixin
+
+class ServiceBusConnectionStringProperties(DictMixin):
+    """
+    Properties of a connection string.
+    """
+    def __init__(self, **kwargs):
+        self.fully_qualified_namespace = kwargs.pop('fully_qualified_namespace', None)
+        self.endpoint = kwargs.pop('endpoint', None)
+        self.entity_path = kwargs.pop('entity_path', None)
+        self.shared_access_signature = kwargs.pop('shared_access_signature', None)
+        self.shared_access_key_name = kwargs.pop('shared_access_key_name', None)
+        self.shared_access_key = kwargs.pop('shared_access_key', None)
+
+
+class ServiceBusConnectionStringParser(object):
```

Comment:
TBH, I'm expecting this to be a module method not a standalone class, or at least a class that has a class/static parse method on it.

Why would the user need to instance an object to parse a conn string? unless this object could be passed around for other usage?

thoughts @KieranBrantnerMagee @rakshith91 @annatisch ?

Reply:
I tend to agree.  While it makes sense to have strongly structured parse output, having the entry point be a function seems like a simpler mental model, in that as Adam says our goal here isn't to present a "broad tool", just to give an imperitive transformation to let users extract properties.  I broadly align with "the fewer new types/classes the better" as well, which may also reveal my leanings on where we'd put the parse method. (I wonder if we could repurpose "from_connection_string", since that's technically what this'll be, or if that carries too much client implicature...)

Reply:
@yunhaoling @KieranBrantnerMagee 
The intention of this parser is NOT to replace the from_connection_string, or not to encourage the use of connection string. 
The goal is to instantiate client types using a code path that adapts to changes in configuration without requiring code to be recompiled.

Please see https://github.com/Azure/azure-sdk-for-python/issues/14574

Here's a detailed explanation by Jesse
https://gist.github.com/jsquire/13742f53bf642ec67baf9491fc844cff

Reply:
as @ramya-rao-a mentioned in the comment, JS approach is to provide a helper method:
[export function parseServiceBusConnectionString](https://github.com/Azure/azure-sdk-for-js/blob/112bb1da8c66ba90655413e3a76c0a899f2d1dee/sdk/servicebus/service-bus/src/util/connectionStringUtils.ts#L53).

I'm thinking that we should do the same thing for python:
```
def parse_servicebus_connection_string(conn_str):
```
so that user just need to call
```python
from azure.servicebus import parse_servicebus_connection_string
conn_str_props = parse_servicebus_connection_string(conn_str)
```
instead of 
```python
from azure.servicebus import ServiceBusConnectionStringParser
parser = ServiceBusConnectionStringParser()
conn_str_props  = parser.parse(conn_str)
```